Arrowgene.Ddon.GameServer/Handler/CraftStartEquipGradeUpHandler.cs

```
@@ -0,0 +1,190 @@
+        {
+            var (storageType, foundItem) = client.Character.Storage.FindItemByUIdInStorage(ItemManager.EquipmentStorages, equipItemUID);
+            if (foundItem != null)
+            {
+                var (slotno, item, itemnum) = foundItem;
+                CharacterCommon characterCommon = null;
+                if (storageType == StorageType.CharacterEquipment || storageType == StorageType.PawnEquipment)
+                {
+                    CurrentEquipInfo.EquipSlot.EquipSlotNo = EquipManager.DetermineEquipSlot(slotno);
+                    CurrentEquipInfo.EquipSlot.EquipType = EquipManager.GetEquipTypeFromSlotNo(slotno);
+                }
+                if (storageType == StorageType.PawnEquipment)
+                {
+                    CurrentEquipInfo.EquipSlot.PawnId = pawnid;
+                    characterCommon = client.Character.Pawns.SingleOrDefault(x => x.PawnId == pawnid);
+                }
+                else if (storageType == StorageType.CharacterEquipment)
+                {
+                    CurrentEquipInfo.EquipSlot.CharacterId = charid;
+                    characterCommon = client.Character;
+                }
+
+                updateCharacterItemNtc.UpdateType = ItemNoticeType.StartEquipGradeUp;
+                updateCharacterItemNtc.UpdateItemList.Add(Server.ItemManager.CreateItemUpdateResult(characterCommon, equipItem, storageType, (byte)slotno, 0, 0));
+
+                _itemManager.UpgradeStorageItem(Server, client, charid, storageType, equipItem, (byte)slotno);
+                updateCharacterItemNtc.UpdateItemList.Add(Server.ItemManager.CreateItemUpdateResult(characterCommon, equipItem, storageType, (byte)slotno, 1, 1));
+                client.Send(updateCharacterItemNtc);
+            }
+            else
+            {
+                Logger.Error($"Item with UID {equipItemUID} not found in {storageType}");
```

We should throw an exception here instead so it doesn't go unnoticed.